tests/utils_tests/test_functional.py

```diff
@@ -225,6 +225,27 @@ def test_lazy_mod_str(self):
         self.assertEqual(lazy_a() % "b", "ab")
         self.assertEqual(lazy_a() % lazy_b(), "ab")
 
+    def test_lazy_mul_int(self):
+        lazy_4 = lazy(lambda: 4, int)
+        lazy_5 = lazy(lambda: 5, int)
+        self.assertEqual(4 * lazy_5(), 20)
+        self.assertEqual(lazy_4() * 5, 20)
+        self.assertEqual(lazy_4() * lazy_5(), 20)
+
+    def test_lazy_mul_list(self):
+        lazy_4 = lazy(lambda: [4], list)
+        lazy_5 = lazy(lambda: 5, int)
+        self.assertEqual([4] * lazy_5(), [4, 4, 4, 4, 4])
+        self.assertEqual(lazy_4() * 5, [4, 4, 4, 4, 4])
+        self.assertEqual(lazy_4() * lazy_5(), [4, 4, 4, 4, 4])
+
+    def test_lazy_mul_str(self):
+        lazy_a = lazy(lambda: "a", str)
+        lazy_5 = lazy(lambda: 5, int)
+        self.assertEqual("a" * lazy_5(), "aaaaa")
+        self.assertEqual(lazy_a() * 5, "aaaaa")
+        self.assertEqual(lazy_a() * lazy_5(), "aaaaa")
```

Comment:
I noticed that these tests work without `__mul__` 🤕 , there is only one assertion that fails without it:
```
    self.assertEqual(lazy_4() * lazy_5(), 20)
```
Can we move others to the first commit?

Reply:
Huh. Am guessing that when not overriding `__mul__` so that `str.__mul__` gets wrapped ends up working because of some internal implementation details, e.g. I guess that the lhs is coerced using `__str__` and the rhs may be cast to `int()`... 🤔 

Anyway, have adjusted the test.